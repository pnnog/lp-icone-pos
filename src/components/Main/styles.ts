import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.mainBg};
  `}
`;

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.white};
  `}
`;
