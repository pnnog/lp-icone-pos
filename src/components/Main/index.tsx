import * as S from './styles';

export type MainProps = {
  title: string;
};

export default function Main({ title }: MainProps) {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
    </S.Wrapper>
  );
}
