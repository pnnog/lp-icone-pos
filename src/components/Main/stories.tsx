import { Story, Meta } from '@storybook/react/types-6-0';
import Main from '.';
import { MainProps } from '.';

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'title default'
  },
  argTypes: {
    title: {
      type: 'string'
    }
  }
} as Meta;

export const Basic: Story<MainProps> = (args) => <Main {...args} />;
Basic.args = {
  title: 'Basic Title'
};

Basic.argTypes = {
  title: {
    type: 'string'
  }
};

export const Secondary: Story<MainProps> = (args) => <Main {...args} />;
Secondary.args = {
  title: 'Secondary Title'
};

Secondary.argTypes = {
  title: {
    type: 'string'
  }
};
