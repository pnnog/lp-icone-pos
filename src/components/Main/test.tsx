import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helper';
import Main from '.';

const props = {
  title: 'Main Title'
};

describe('Main', () => {
  it('should render main title', () => {
    renderWithTheme(<Main {...props} />);
    expect(
      screen.getByRole('heading', {
        name: /Main Title/i
      })
    ).toBeInTheDocument();
  });

  it('should render title with white color', () => {
    renderWithTheme(<Main {...props} />);
    expect(screen.getByRole('heading', { name: props.title })).toHaveStyle({
      color: '#FAFAFA'
    });
  });

  it('should render main with bg color', () => {
    const { container } = renderWithTheme(<Main {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }).parentNode
    ).toHaveStyle({
      background: '#0000FF'
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
