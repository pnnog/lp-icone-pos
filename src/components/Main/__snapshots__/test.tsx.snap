// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render main with bg color 1`] = `
.c0 {
  background-color: #0000FF;
}

.c1 {
  color: #FAFAFA;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Main Title
  </h1>
</div>
`;
